packages/aws-cdk/lib/cli/cdk-toolkit.ts

```diff
@@ -1218,30 +1211,28 @@ export class CdkToolkit {
   }
 
   public async refactor(options: RefactorOptions): Promise<number> {
-    if (!options.dryRun) {
-      info('Refactor is not available yet. Too see the proposed changes, use the --dry-run flag.');
-      return 1;
+    let exclude: string[] = [];
+    if (options.excludeFile != null) {
+      if (!(await fs.pathExists(options.excludeFile))) {
+        throw new ToolkitError(`The exclude file ${options.excludeFile} does not exist`);
+      }
+      exclude = fs.readFileSync(options.excludeFile).toString('utf-8').split('\n');
```

Comment:
Not super important, but this could probably done with a single "read" operation and catching the `ENOENT` error.